### PR TITLE
fix: workqueue need to retry when failed

### PR DESCRIPTION
**Pull request by gxthrj**
Need to retry when failed.

### CODE REVIEW

**Comment by tokers** [comment]
@gxthrj Please add some explanations to show the purpose you submitted this PR.

**Comment by gxthrj** [comment]
We need to retry when sync CRD  failed.
Use `ItemFastSlowRateLimiter` to keep retrying until it succeeds or a new configuration overwrites the previous outdated configuration.

**Comment by gxthrj** [comment]
@tokers I just found the const `_maxRetries`, It is unreasonable to specify the number of retries because the retries will end in a short time.

**Comment by gxthrj** [comment]
BTW, when trigger retry, we should warn, not `info` level log.
Will modify in this PR, too.

**Comment by tokers** [comment]
> BTW, when trigger retry, we should warn, not `info` level log.
> Will modify in this PR, too.

OK.

**Comment by tokers** [comment]
> We need to retry when sync CRD failed.
> Use `ItemFastSlowRateLimiter` to keep retrying until it succeeds or a new configuration overwrites the previous outdated configuration.

It looks this limiter has an exponential back off? It so, that's would be great.

**Comment by tokers** [comment]
> @tokers I just found the const `_maxRetries`, It is unreasonable to specify the number of retries because the retries will end in a short time.

I referred to the implementation endpoints controller.

File: pkg/ingress/controller/apisix_route.go [hunk 1]
```diff
@@ -188,15 +188,11 @@ func (c *apisixRouteController) handleSyncErr(obj interface{}, err error) {
 		c.workqueue.Forget(obj)
 		return
 	}
-	if c.workqueue.NumRequeues(obj) < _maxRetries {
```
Comment (tokers) [inline pkg/ingress/controller/apisix_route.go:192]: See the use pattern in https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/endpoint/endpoints_controller.go#L356.

We just reference it.

Comment (gxthrj) [reply pkg/ingress/controller/apisix_route.go:192]: I suggest to cancel the maximum number of times limit first, and add it when our admission and crd status are improved. Now we need real-time error reporting to quickly solve the problem.

Comment (tokers) [reply pkg/ingress/controller/apisix_route.go:192]: I cannot claim a distinct point about this now. Just merge it if it makes some trouble at present.

**Review by tokers — APPROVED** [review]

File: pkg/ingress/controller/apisix_route.go (hunk 1 above)
Comment (gxthrj) [reply pkg/ingress/controller/apisix_route.go:192]: OK

**Pull request merged**